### PR TITLE
Develop dpd

### PR DESCRIPTION
**Pull request by takaya-meno-uhuru**
add the docs of the file deployment

### CODE REVIEW

File: en/FileDeploys/index.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+---
+lastUpdated: 2018-10-24
+---
+
+# File Deploys
+
+The **File Deploys** deploys files to enebular agent.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: **File Deploys** deploys files to enebular agent. 
(Remove the)

File: en/FileDeploys/index.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
+---
+lastUpdated: 2018-10-24
+---
+
+# File Deploys
+
+The **File Deploys** deploys files to enebular agent.  
+The use cases, for example, running shell script in enebular agent, adding setting files and so on.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: A use case example can be running shell script in enebular agent or adding setting files.

File: en/FileDeploys/index.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
+---
+lastUpdated: 2018-10-24
+---
+
+# File Deploys
+
+**File Deploys** deploys files to enebular agent.  
+A use case example can be running shell scripts in enebular agent or adding setting files.
+
+Files are kind of asset and included as follows.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: A file is an asset with one of the following possible functionalities.

File: en/FileDeploys/index.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+- execution files for enebular agent
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: executing files for enebular agent

File: en/FileDeploys/index.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+- setting files for enebular agent
+- other files for enebular agent (image files, movie files)
+
+Owner/Admin of the project are able to deploy files. Collaborator can't do that.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: Owner/Admin of the project are able to deploy files. Collaborators are unable to.

File: en/FileDeploys/index.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+To prevent from untrustable deployment, it's only allowed trustable user to deploy it.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: Deployments can only be made by trusted users.

File: en/FileDeploys/index.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+## Screen description
+### List
+
+The list displays uploaded files.
+Selecting a file, you can view the detail of the file.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: Select a file to view it's details.

File: en/FileDeploys/index.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+See **File settings** for further information about config.
+
+### Overview
+
+You can see general information about the uploaded files.
+
+### Deploy
+
+You can deploy files.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: You can deploy files and view deployment history.

File: en/FileDeploys/index.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+Also, see the history of the deployments.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: (used in sentence above)

File: en/FileDeploys/index.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+### Devices
+
+You can delete files which deployed in devices.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: You can delete files which are deployed in devices.

File: en/FileDeploys/index.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+
+You can set the path to deploy.
+
+- path under `(root)/assets/`
+- folder name and file name
+- e.g. Setting hoge/hige.txt, saving as (root)/assets/hoge/hige.txt.
+
+### Execution
+
+Setting for you want to deploy execution files
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: Setting for if you want to deploy execution files.

File: en/FileDeploys/index.md [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+
+#### Execute On Deploy
+
+If it's turned on, files are run as the execution files.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: If it's turned on, files run as the execution files.

File: en/FileDeploys/index.md [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+The user is same as enebular-runtime-agent. 
+
+#### Max Execution Time
+
+You can set maximum execution time of **Execution**.
+It's able to set from 0 to 300(5 min)sec.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: Set times are from 0 to 300(5 min)sec.

File: en/FileDeploys/index.md (hunk 3 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 4 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 5 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 6 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 7 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 8 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 9 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 10 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 12 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 13 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+Setting environmental variables
+
+- KEY
+    - the name of enviromnetal variable
+- VALUE
+    - the value of enviromnetal variable
+
+### Deploy Hooks
+
+It's a functionality that you can execute files already deployed when you deploy new files.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: It's a functionality that allows you can execute files already deployed when you deploy new files.

File: en/FileDeploys/index.md [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+
+#### Staging
+
+##### Pre-Deploy
+
+You can execute a file in enebular agent before the deployment as Pre-Deploy.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: Pre-Deploy executes a file in enebular agent before a deployment.

File: en/FileDeploys/index.md [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+If you don't specify it, Pre-Deploy doesn't run.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: If you don't specify Pre-Deploy it doesn't run.

File: en/FileDeploys/index.md [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+
+e.g.) Before deployment, you want to run shell script which deleting some files you don't need anymore in enebular agent.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: e.g.)You want to run a shell script before deployment that deletes some files you don't need anymore in enebular agent.

File: en/FileDeploys/index.md [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+
+##### Post-Deploy
+
+You can execute a file in enebular agent after the deployment as Post-Deploy.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: Post-Deploy executes a file in enebular agent after deployment.

File: en/FileDeploys/index.md [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+If you don't specify it, Post-Deploy doesn't run.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: If you don't specify Post-Deploy it doesn't run.

File: en/FileDeploys/index.md [hunk 20]
```diff
@@ -0,0 +1,147 @@
...
+
+e.g.) After deployment, you want to run shell script which deleting some files you don't need anymore in enebular agent.
```
Comment (Mantis-Ranson) [inline en/FileDeploys/index.md]: e.g.)  You want to run a shell script after deployment which deletes some files you don't need anymore in enebular agent.

File: en/FileDeploys/index.md (hunk 14 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 15 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 16 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 17 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 18 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 19 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

File: en/FileDeploys/index.md (hunk 20 above)
Comment (tkymeno2) [reply en/FileDeploys/index.md]: なおしました

**Comment by tkymeno2** [comment]
@k-harada413 そうです。

**Pull request merged**